src/payout_tokens.cc

```diff
@@ -0,0 +1,183 @@
+#include "base/values.h"
+
+using namespace std::placeholders;
+
+namespace confirmations {
+
+PayoutTokens::PayoutTokens(
+    ConfirmationsImpl* confirmations,
+    ConfirmationsClient* confirmations_client) :
+    public_key_(nullptr),
+    confirmations_(confirmations),
+    confirmations_client_(confirmations_client) {
+  BLOG(INFO) << "Initializing payout tokens";
+}
+
+PayoutTokens::~PayoutTokens() {
+  BLOG(INFO) << "Deinitializing payout tokens";
+}
+
+void PayoutTokens::Payout(
+    const WalletInfo& wallet_info,
+    const std::string& public_key) {
+  payment_id_ = wallet_info.payment_id;
+
+  public_key_ = PublicKey::decode_base64(public_key);
+
+  RedeemPaymentTokens();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void PayoutTokens::RedeemPaymentTokens() {
+  BLOG(INFO) << "RedeemPaymentTokens";
+
+  auto unblinded_payment_tokens = confirmations_->GetUnblindedPaymentTokens();
+  if (unblinded_payment_tokens.size() == 0) {
+    BLOG(ERROR) << "No unblinded payment tokens";
+
+    return;
+  }
+
+  // Create payload
+  base::DictionaryValue payload;
+  payload.SetKey("paymentId", base::Value(payment_id_));
+
+  std::string payload_json;
+  base::JSONWriter::Write(payload, &payload_json);
+
+  // Create payment credentials
+  auto payment_credentials_list = std::make_unique<base::ListValue>();
+  for (auto& unblinded_payment_token : unblinded_payment_tokens) {
+    // Create credential
+    base::DictionaryValue credential_dictionary;
+
+    auto verification_key = unblinded_payment_token.derive_verification_key();
+
+    auto verification_signature = verification_key.sign(payload_json);
+    auto verification_signature_base64 = verification_signature.encode_base64();
+    credential_dictionary.SetKey("signature",
+        base::Value(verification_signature_base64));
+
+    auto preimage = unblinded_payment_token.preimage();
+    auto preimage_base64 = preimage.encode_base64();
+    credential_dictionary.SetKey("t", base::Value(preimage_base64));
+
+    // Create payment credential
+    auto* payment_credential_dictionary = new base::DictionaryValue();
+
+    payment_credential_dictionary->SetKey("credential",
+        base::DictionaryValue(std::move(credential_dictionary)));
+
+    auto public_key_base64 = public_key_.encode_base64();
+    payment_credential_dictionary->SetKey("publicKey",
+        base::Value(public_key_base64));
+
+    payment_credentials_list->Append(
+        std::unique_ptr<base::DictionaryValue>(payment_credential_dictionary));
+  }
+
+  // Create request body
+  base::DictionaryValue body_dictionary;
+
+  body_dictionary.SetWithoutPathExpansion("paymentCredentials",
+      std::move(payment_credentials_list));
+
+  body_dictionary.SetKey("payload", base::Value(payload_json));
+
+  std::string json;
+  base::JSONWriter::Write(body_dictionary, &json);
+  std::string body = json;
+
+  // Create request headers
+  std::vector<std::string> headers = {};
```

Comment:
Does explicitly initializing this serve any purpose?

Reply:
a habit, all fixes throughout codebase 👍